ros/angel_system_nodes/angel_system_nodes/hl2ss_ros_bridge.py

```diff
@@ -0,0 +1,316 @@
+
+        # Declare ROS topics
+        self._image_topic = (
+            self.declare_parameter("image_topic", "")
+            .get_parameter_value()
+            .string_value
+        )
+        self._hand_pose_topic = (
+            self.declare_parameter("hand_pose_topic", "")
+            .get_parameter_value()
+            .string_value
+        )
+        self.ip_addr = (
+            self.declare_parameter("ip_addr", "")
+            .get_parameter_value()
+            .string_value
+        )
+
+        log = self.get_logger()
+        if self._image_topic == "":
+            raise ValueError("Please provide the image topic with the `image_topic` parameter")
+        if self._hand_pose_topic == "":
+            raise ValueError("Please provide the hand pose topic with the `hand_pose_topic` parameter")
+        if self.ip_addr == "":
+            raise ValueError("Please provide HL2 IPv4 address with the `ip_addr` parameter")
+
+
+        log.info(f"Image topic: {self._image_topic}")
+        log.info(f"Hand pose topic: {self._hand_pose_topic}")
+        log.info(f"HL2 IP: {self.ip_addr}")
+
+        # Define HL2SS server ports
+        self.pv_port = hl2ss.StreamPort.PERSONAL_VIDEO
+        self.si_port = hl2ss.StreamPort.SPATIAL_INPUT
+
+        # Create frame publisher
+        self.frame_publisher = self.create_publisher(
+            Image,
+            self._image_topic,
+            1
+        )
+        # Create the hand joint pose publisher
+        self.hand_publisher = self.create_publisher(
+            HandJointPosesUpdate,
+            self._hand_pose_topic,
+            1
+        )
+
+        log.info("Connecting to HL2SS servers...")
+        self.connect_hl2ss_pv()
+        log.info("PV client connected!")
+        self.connect_hl2ss_si()
+        log.info("SI client connected!")
+
+        log.info("Starting publishing threads...")
+        # Start the frame publishing thread
+        self._pv_active = Event()
+        self._pv_active.set()
+        self._pv_thread = Thread(
+            target=self.pv_publisher,
+            name="publish_frames"
+        )
+        self._pv_thread.daemon = True
+        self._pv_thread.start()
+        # Start the hand tracking data thread
+        self._si_active = Event()
+        self._si_active.set()
+        self._si_thread = Thread(
+            target=self.si_publisher,
+            name="publish_frames"
+        )
+        self._si_thread.daemon = True
+        self._si_thread.start()
+        log.info("Starting publishing threads... Done")
+
+
+    def connect_hl2ss_pv(self) -> None:
+        """
+        Creates the HL2SS PV client and connects it to the server on the headset.
+        """
+        # Operating mode
+        # 0: video
+        # 1: video + camera pose
+        # 2: query calibration (single transfer)
+        mode = hl2ss.StreamMode.MODE_1
+
+        # Camera parameters
+        width     = 1280
+        height    = 720
+        framerate = 30
```

Comment:
This is fine for now, but we could expose these as node parameters with these values as defaults.